src/psyclone/parse/kernel.py

```diff
@@ -612,6 +634,39 @@ def get_kernel_metadata(name, ast):
     return ktype
 
 
+def get_kernel_interface(ast):
```

Comment:
We have a choice here. We can either mandate that (for LFRic Kernels) there must be either a type-bound procedure or a *single* interface, OR, we mandate that if there is an interface then it must be named `<kernel_name>_code`. The latter would allow the Kernel file to contain more than one interface should that ever be needed. If we go for the latter, then this routine will need to be passed the name of the kernel (which I've in fact suggested later for the benefit of the error message). Worth checking what @TeranIvy and @rupertford think?

Reply:
I don't believe having multiple interfaces for the kernel is a good idea at all. We won't be doing that in LFRic. That implies the former which the code already does.